src/deriver/ScopedKeys.js

```diff
@@ -74,7 +84,7 @@ class ScopedKeys {
         scope: options.identifier,
       };
 
-      this._deriveHKDF(saltBuf, Buffer.concat([inputKeyBuf, keyRotationSecretBuf]), contextBuf, KEY_LENGTH)
+      return this._deriveHKDF(saltBuf, Buffer.concat([inputKeyBuf, keyRotationSecretBuf]), contextBuf, KEY_LENGTH)
```

Comment:
Is it possible for either `_deriveLegacySyncKey` or `_deriveHKDF` to fail? If so, are their failures propagated?

I ask because the block starting on line L41 is wrapped in a `new Promise`, I think the promise returned by `this._deriveHKDF` will be ignored and only the `resolve` on line L86 will be honored, but I haven't used a combination of `return` and `resolve` before.

Reply:
AFAIK you're correct that only the `resolve` will be honoured; I added the `return` on this line because the `return resolve()` added above as making the linter complain about "inconsistent return from function".

I would not expect their of these to ever fail, but I suppose it's possible around the edges, e.g. failing to decode a hex buffer or in an environment here `sha256` is not available for some reason.  It's probably cleaner if we do a consistent return of errors from the promises even if we never expect them.  I'll update.

Reply:
I'm going to address this by doing `return resolve(this._deriveHKDF(...).then(...))` which will propagate any rejections from the underlying promise.